### PR TITLE
[3/N] Fix extra warnings brought by clang-tidy-17

### PR DESCRIPTION
**Pull request by cyyever**
Follows #137459

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/csrc/inductor/aoti_torch/shim_common.cpp [hunk 1]
```diff
@@ -47,29 +47,26 @@
...
+#else
+#include <filesystem>
+namespace fs = std::filesystem;
 #endif
 
 // HACK for failed builds in ARVR, where it cannot find these symbols within
 // std::experimental::filesystem
 namespace {
 std::string get_current_path() {
-#if __has_include("filesystem") && !defined(__linux__)
```
Comment (ezyang) [inline torch/csrc/inductor/aoti_torch/shim_common.cpp:69]: I'm not sure if this will work internally unfortunately

**Review by ezyang — APPROVED** [review]

File: torch/csrc/inductor/aoti_torch/shim_common.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/inductor/aoti_torch/shim_common.cpp:69]: The branch is for Windows.

**Comment by cyyever** [comment]
@ezyang Does it work in META?

**Pull request closed without merging**